Bump min sdk version to 24

xcho1 opened the pull request:
bumped minsdk version to 24 to support display config changes

xcho1 committed:
bump min sdk version to 24

dbnicholson commented on .p4a:
```diff
@@ -7,7 +7,7 @@
 --requirements python3,android,pyjnius,genericndkbuild,sqlite3,cryptography,twisted,attrs,bcrypt,service_identity,pyasn1,pyasn1_modules,pyopenssl,openssl,six,liblzma,zstandard,beautifulsoup4==4.10.0,zimply_core,kolibri_explore_plugin~=3.0,kolibri_zim_plugin
 --android-api 30
---minsdk 21
+--minsdk 24
```
I think you're going to have to add `--allow-minsdk-ndkapi-mismatch` to this file to avoid some bogus advice from p4a. If you look at the NDK 23 [changelog](https://github.com/android/ndk/wiki/Changelog-r23), you can see that it supports all the way back to API 16. I don't know why p4a suggests 21 as the minimum NDK API.

dbnicholson replied:
Alternatively, you could add `--ndk-api 24`, I believe. I actually don't get the point of the "NDK API" value at all. The only way it affects the build is the version checking guards that p4a does.

xcho1 committed:
add ndk ignore flag

The p4a guard this disables is bogus. Each NDK has a specific set of SDK
versions it supports and there's actually no such thing as the "NDK
API". If the minimum SDK version is set below what the NDK supports, it
will just fail to build.

dbnicholson reviewed (approved):
Looks good. I'm adding some context to the `--allow-minsdk-ndkapi-mismatch` commit message and then I'll merge.
